cas-server-support-saml/src/main/java/org/jasig/cas/support/saml/authentication/principal/SamlService.java

```diff
@@ -183,13 +178,18 @@ protected static String extractRequestId(final String requestBody) {
     protected static String getRequestBody(final HttpServletRequest request) {
         final StringBuilder builder = new StringBuilder();
         try (final BufferedReader reader = request.getReader()) {
+
+            if (reader == null) {
+                LOGGER.debug("Request body could not be read because it's empty.");
+                return null;
+            }
             String line;
             while ((line = reader.readLine()) != null) {
                 builder.append(line);
             }
             return builder.toString();
         } catch (final Exception e) {
-            LOGGER.error("Could not obtain the saml request body from the http request", e);
+            LOGGER.trace("Could not obtain the saml request body from the http request", e);

```

Comment:
It's going to produce too much noise, as not all saml requests contain a body. Specially the initial one read by the extractor will only have url parameters and setting this to debug will log a nasty exception that may confuse deployers.